types/react-leaflet/index.d.ts

```diff
@@ -196,13 +196,13 @@ export interface TileLayerProps extends TileLayerEvents, Leaflet.TileLayerOption
     children?: Children;
     url: string;
 }
-export class TileLayer<P extends TileLayerProps, E extends Leaflet.TileLayer> extends GridLayer<P, E> { }
+export class TileLayer<P = TileLayerProps, E extends Leaflet.GridLayer = Leaflet.TileLayer> extends GridLayer<P, E> { }
```

Comment:
I'm confused about the constraint on `E`. It's true that it should be a subtype of `Leaflet.GridLayer`, but I would think it should also be a subtype of `Leaflet.TileLayer`. That is, any class that extends this one should provide a type parameter `E` that is a subtype of `Leaflet.TileLayer` (with the understanding that any type is a subtype of itself, so `E` could in fact be `Leaflet.TileLayer`). I'm also confused because it seems inconsistent to impose a type constraint for `E` here (and also in `TileLayer`, `WMSTileLayer`, `LayersControl`, and `ZoomControl`) when no such constraint is used for the other classes or for the other type parameter, `P`. I suggest either:
(1) impose constraints on all type parameters for all classes and use the same type for the constraint as for the default type parameter, for example:
```typescript
export class Pane<P extends PaneProps = PaneProps, S extends PaneState = PaneState> extends React.Component<P, S> { ... }
export class TileLayer<P extends TileLayerProps = TileLayerProps, E extends Leaflet.TileLayer = Leaflet.TileLayer> extends GridLayer<P, E> { }
```
or
(2) remove the constraints and just leave the default type parameters, for example:
```typescript
export class TileLayer<P = TileLayerProps, E = Leaflet.TileLayer> extends GridLayer<P, E> { }
export class WMSTileLayer<P = WMSTileLayerProps,  E = Leaflet.TileLayer.WMS> extends GridLayer<P, E> { }
```
What do you think?